### PR TITLE
Component overlays

### PR DESCRIPTION
**Pull request by btd90**
Components can now be placed on the map via two 'helper' components, allowing either a single lat/lng location to be specified (for which any children will render from using the specified point as the top right corner) OR specifying a bounding box to house any child components (which will result in a fixed size for child components to adhere to)

### CODE REVIEW

File: app/pods/components/map-component/display-box/component.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        let childNodes = pane.childNodes;
+        let found = false;
+
+        // Check if child elements already exist
+        if(childNodes && !found) {
+          // Update position/visibility if child node exists
+          for(let i=0; i<childNodes.length; i++) {
+            if(childNodes[i].id === layerId) {
+              this._update();
+              found = true;
```
Comment (ultish) [inline app/pods/components/map-component/display-box/component.js:35]: break?

File: app/pods/components/map-component/display-box/component.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    // Add layer to relevant layer group
+    L.customlayer({
+      layerId: layerId
+    }).addTo(layerGroup);
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.set('layerReady', true);
```
Comment (ultish) [inline app/pods/components/map-component/display-box/component.js:109]: is layerReady used internally somewhere?

File: app/pods/components/map-component/display-point/component.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        let childNodes = pane.childNodes;
+        let found = false;
+
+        // Check if child elements already exist
+        if(childNodes && !found) {
+          // Update position/visibility if child node exists
+          for(let i=0; i<childNodes.length; i++) {
+            if(childNodes[i].id === layerId) {
+              this._update();
+              found = true;
```
Comment (ultish) [inline app/pods/components/map-component/display-point/component.js:34]: break?

File: app/pods/home/editor/template.hbs [hunk 4]
```diff
@@ -2,20 +2,115 @@
...
+            componentOverlays=componentOverlays}}
+          {{#if layerGroup}}
+            {{!-- Display point to render child components on the map at the given lat/lng --}}
+            {{#map-component/display-point
+                layerId=firstLayerId
+                lat=firstLayerLatitude
+                lng=firstLayerLongitude
+                mapInstance=mapInstance
+                layerGroup=layerGroup
+                layerReady=layerReady}}
```
Comment (ultish) [inline app/pods/home/editor/template.hbs:39]: how does this layerReady work? was display-point meant to yield layerReady so you could use it in a block component here? eg

```
{{#map-component/display-point 
    ...
as |dp|}}

   {{#if dp.layerReady}}
      ...
   {{/if}}
{{/map-component/display-point }}
```

File: app/pods/components/svg-component/component.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import Component from '@ember/component';
+
+/**
+ * SVG component for visualisation on the map.
+ */
+export default Component.extend({
+    tagName: '',
```
Comment (ultish) [inline app/pods/components/svg-component/component.js:7]: fyi the styles.scss in this folder won't do anything as this component is tag less :)

File: app/pods/components/map-component/display-box/component.js (hunk 1 above)
Comment (btd90) [reply app/pods/components/map-component/display-box/component.js:35]: Yep will update

File: app/pods/components/map-component/display-point/component.js (hunk 3 above)
Comment (btd90) [reply app/pods/components/map-component/display-point/component.js:34]: Will update

File: app/pods/home/editor/template.hbs (hunk 4 above)
Comment (btd90) [reply app/pods/home/editor/template.hbs:39]: That was my intention, I'll update to use this cleaner approach

File: app/pods/components/svg-component/component.js (hunk 5 above)
Comment (btd90) [reply app/pods/components/svg-component/component.js:7]: Will remove

File: app/pods/components/map-component/display-box/component.js (hunk 2 above)
Comment (btd90) [reply app/pods/components/map-component/display-box/component.js:109]: Only for child components, will update code block based on your other comment

**Pull request merged**